### DISCUSSION TITLE
How to totate the whole display by 90 degrees?

### DISCUSSION THREAD

**Discussion by harshitpatel96** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1782] [Q&A]
Hi everyone,

First of all let me clear that I am not a C++ coder, but I have basic understanding of it.

I want to rotate the UI by 90 degree, I was looking at the code, and I found a flag which when enabled would rotate the screen by 180 degree. Is there any way I can change the orientation to 90 degree and make it vertical?

**Comment by oldman4U** [comment]
No. Because this would require a complete redesign of the current screen layout.

**Comment by harshitpatel96** [comment]
@oldman4U Can you point me to some documentation on how can I achieve this (both changing the screen layout and a vertical orientation)? Also how big a challenge this is?

**Comment by oldman4U** [comment]
Unfortunately I believe such a documentation does not exist. You will have to find out by yourself by digging into the code. Sorry

**Comment by 9zdot** [comment]
Hi all,
I'm also not a C++ developer but I'm trying to achieve the same objective.
I was able to rotate the screen, fix the touch to work with the portrait orientation but I'm still failing with the layout. It's a pretty mess right now. 
![btt70_portrait](https://user-images.githubusercontent.com/5232795/114053589-752e0080-9865-11eb-9d57-c225c8d490e8.jpg)

In the actual situation, I'm able to use the original  screen setting submenu to rotate the screen in 360 degrees, rotating 90 degrees clockwise on each step.

The way that was developed is very stupid, I mean, they didn't take consideration to have portrait feature, so a deep refactoring is needed.

I'm working on this feature in my rare free time, so I don't have any idea when a PR will be available.

**Reply by victornpb** [reply]
do you have any progress on it?

**Reply by 9zdot** [reply]
Unfortunately no.
I had no more time to play around with this code.
